packages/trailbase-db-collection/src/trailbase.ts

```diff
@@ -285,9 +322,44 @@ export function trailBaseCollectionOptions<
       }
 
       start()
+
+      // If we're in on-demand mode, expose loadSubset/unloadSubset handlers
+      if (internalSyncMode === `eager`) {
+        return
```

Comment:
Looking at electric, it seems to do the opposite, i.e. fetch erverything in eager mode: https://github.com/TanStack/db/blob/8ef8f09bc8df6bb49908150818528f10b46d553a/packages/electric-db-collection/src/electric.ts#L386